listenbrainz_spark/request_consumer/request_consumer.py

```diff
@@ -103,6 +103,7 @@ def push_to_result_queue(self, messages):
     def callback(self, channel, method, properties, body):
         request = json.loads(body.decode('utf-8'))
         logger.info('Received a request!')
+
         while True:
             try:
                 self.request_channel.basic_ack(delivery_tag=method.delivery_tag)
```

Comment:
I went down the rabbit hole of the rabbitmq timeouts etc. and looks like this is problematic. RabbitMQ mandates that the `basic_ack` be sent from the same channel that received the message. So if we get an error during acking, it's already too late. There's no point of retrying.

Reply:
https://github.com/metabrainz/listenbrainz-server/blob/b3fec9f9ea9294a69b0c588a60e3ecf6773559f8/listenbrainz_spark/request_consumer/request_consumer.py#L77-L91

This is an even bigger issue here. If we encounter an error during publishing the results, we close the connection and create new connections and channels for both consuming and publishing. This is erroneous because the original channel which received the request got closed so we cannot ack the request later. We should just be recreating the result channel here.